internal/catalogtask/catalogtask.go

```diff
@@ -0,0 +1,122 @@
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/common"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/logger"
+)
+
+type TaskOperator interface {
+	Get() (*common.RequestMessage, error)
+	Update(data map[string]interface{}) error
+}
+
+type CatalogTask struct {
+	url  string
+	ctx  context.Context
+	glog logger.Logger
+}
+
+func MakeCatalogTask(ctx context.Context, url string) *CatalogTask {
+	glog := logger.GetLogger(ctx)
+
+	return &CatalogTask{ctx: ctx, url: url, glog: glog}
+}
+
+func (ct *CatalogTask) Get() (*common.RequestMessage, error) {
+	body, err := getWorkPayload(ct.glog, ct.url)
+	if err != nil {
+		ct.glog.Errorf("Error reading payload in %s %v", ct.url, err)
+		return nil, err
+	}
+
+	req, err := parseRequest(ct.glog, body)
+	if err != nil {
+		ct.glog.Errorf("Error parsing payload in %s %v", ct.url, err)
+	}
+	return req, err
+}
+
+func getWorkPayload(glog logger.Logger, url string) ([]byte, error) {
+	client := &http.Client{}
+	xrh := "eyJlbnRpdGxlbWVudHMiOnsiaW5zaWdodHMiOnsiaXNfZW50aXRsZWQiOnRydWUsImlzX3RyaWFsIjpmYWxzZX0sImNvc3RfbWFuYWdlbWVudCI6eyJpc19lbnRpdGxlZCI6dHJ1ZSwiaXNfdHJpYWwiOmZhbHNlfSwibWlncmF0aW9ucyI6eyJpc19lbnRpdGxlZCI6dHJ1ZSwiaXNfdHJpYWwiOmZhbHNlfSwiYW5zaWJsZSI6eyJpc19lbnRpdGxlZCI6dHJ1ZSwiaXNfdHJpYWwiOmZhbHNlfSwidXNlcl9wcmVmZXJlbmNlcyI6eyJpc19lbnRpdGxlZCI6dHJ1ZSwiaXNfdHJpYWwiOmZhbHNlfSwib3BlbnNoaWZ0Ijp7ImlzX2VudGl0bGVkIjp0cnVlLCJpc190cmlhbCI6ZmFsc2V9LCJzbWFydF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlLCJpc190cmlhbCI6ZmFsc2V9LCJzdWJzY3JpcHRpb25zIjp7ImlzX2VudGl0bGVkIjp0cnVlLCJpc190cmlhbCI6ZmFsc2V9LCJzZXR0aW5ncyI6eyJpc19lbnRpdGxlZCI6dHJ1ZSwiaXNfdHJpYWwiOmZhbHNlfX0sImlkZW50aXR5Ijp7ImludGVybmFsIjp7ImF1dGhfdGltZSI6Nzk5LCJvcmdfaWQiOiIxMTc4OTc3MiJ9LCJhY2NvdW50X251bWJlciI6IjYwODk3MTkiLCJhdXRoX3R5cGUiOiJiYXNpYy1hdXRoIiwidXNlciI6eyJpc19hY3RpdmUiOnRydWUsImxvY2FsZSI6ImVuX1VTIiwiaXNfb3JnX2FkbWluIjp0cnVlLCJ1c2VybmFtZSI6Imluc2lnaHRzLXFhIiwiZW1haWwiOiJkYWpvaG5zb0ByZWRoYXQuY29tIiwiZmlyc3RfbmFtZSI6Ikluc2lnaHRzIiwidXNlcl9pZCI6IjUxODM0Nzc2IiwibGFzdF9uYW1lIjoiUUEiLCJpc19pbnRlcm5hbCI6dHJ1ZX0sInR5cGUiOiJVc2VyIn19"
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		glog.Errorf("Error creating request %s %v", url, err)
+		return nil, err
+	}
+	req.Header.Add("x-rh-identity", xrh)
+	resp, err := client.Do(req)
+	if err != nil {
+		glog.Errorf("Error fetching request %s %v", url, err)
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	return ioutil.ReadAll(resp.Body)
+}
+
+// Parse the request into RequestMessage
+func parseRequest(glog logger.Logger, b []byte) (*common.RequestMessage, error) {
+	req := common.RequestMessage{}
+	decoder := json.NewDecoder(bytes.NewReader(b))
+	decoder.UseNumber()
+	err := decoder.Decode(&req)
+	if err != nil {
+		glog.Errorf("Error decoding json %v", err)
+		return nil, err
+	}
+	return &req, nil
+}
+
+func (ct *CatalogTask) Update(data map[string]interface{}) error {
```

Comment:
@bzwei Do you want to move this below the Get so all exported functions are together